src/compiler/preprocess/index.ts

```diff
@@ -83,6 +83,82 @@ async function replace_async(
 	return out.concat(final_content);
 }
 
+
+
+/**
+ * import decoded sourcemap from mozilla/source-map/SourceMapGenerator
+ * forked from source-map/lib/source-map-generator.js
+ * from methods _serializeMappings and toJSON
+ */
+function decodedSourcemapFromSourceMapGenerator(this: any) {
```

Comment:
It seems a bit strange to me to use a `this` variable instead of just making it `generator` or something and calling the function normally instead of via `apply`

Reply:
agree
the idea was: keep the functions similar to the original fns in mozilla/source-map
but there is no real benefit